db/controllers/articles-controller.js

```diff
@@ -1,12 +1,26 @@
-const { readArticleByID } = require("../models/articles-model");
+const { readArticles, readArticleByID } = require("../models/articles-model");
+const { readComments } = require("../models/comments-models");
+
+const getArticles = (req, res, next) => {
+  const promises = [readArticles(), readComments()];
+  return Promise.all(promises).then(([articles, comments]) => {
+    articles.forEach((article) => {
+      const articleComments = comments.filter((comment) => {
+        return comment.article_id === article.article_id;
+      });
+      article.comment_count = articleComments.length
```

Comment:
This is cleverly done, however, we shouldn't be doing this sort of thing in the controller. We can make a single query in the articles model that gets all of the articles, then joins the table with the comments one, to create the comment_count. 
We can then just invoke the articles model here, which will give us all the required information. 👍